### ISSUE TITLE
Error: Failed to find 'delve' command line tool on Ubuntu 18

### ISSUE THREAD

**Issue by photomedia** [eprints/eprints3.4#167]
This line:
`my $delve_location = "";`
leads to an error (testing on EPrints 3.4.3 on Ubuntu 18) https://github.com/eprints/eprints3.4/blob/dcf4f7f61afe9a83a395954e22cf5288206b1bad/bin/check_xapian#L189

The script returns the following error as a result of running check_xapian REPOID --reindex:
`"Error: Failed to find 'delve' command line tool"`
The error appears even if the delve tool is there, it's the script that fails to find it.
Modifying the double quotes in this line so that they are single quotes fixes the issue.  Changing the above initialization to the following:
`my $delve_location = '';`
solves the issue.  I think it must be something about the string types and comparison in the line:
https://github.com/eprints/eprints3.4/blob/dcf4f7f61afe9a83a395954e22cf5288206b1bad/bin/check_xapian#L200

**Comment by drn05r** [comment]
I was not able to replicate this when I first tested this on an exisiting Ubuntu 18.04 server but I need to setup a proper Ubuntu 18.04 EPrints test installation so I can investigate this in a vanilla state.

**Comment by drn05r** [comment]
I have now thoroughly tested this on Ubuntu18.04 by installing EPrints 3.4.3 from its Deb package and I was unable to reproduce this.  I have never seen an interpolated empty string (double quotes) be latterly interpreted differently to a non-interpolated empty string (single quotes), so something must have changed in the intervening period of your testing.  

One issue may have been that xapian-tools DEB package may not have been installed.  However, (and as I found), this would complain about **quest** command not being found and exit before it even tries for **delve**.

I have spotted a couple of things that need tidying up, as a result of introducing support for facets.
1. A missing newline (\n) on output on line 490.
2. A check for whether the year is indexed is only applicable for facets where year in stored as a facet separate to the full date

**Comment by photomedia** [comment]
My apologies for the confusing error report on my parts.  In the end, the issue I had was caused by what I think was an out-of-date Xapian index.  The folder structure under /REPOID/var/xapian did not have the iamglass file, while the server only had "xapian-delve" and no "delve".    That combination caused this error, as it is unexpected in the script, which has this line:

`$delve_location = `/usr/bin/which xapian-delve` if -e $path."/iamglass";`

The error went away after I did epadmin reindex REPOID.  The /REPOID/var/xapian/ folder was regenerated with the expected "iamglass" file present, so the error went away.
